app/models/admin_setting.rb

```diff
@@ -51,7 +53,8 @@ def self.guest_downloading_off?
     self.first ? self.first.guest_downloading_off? : false
   end
   def self.default_skin
-    self.first ? (self.first.default_skin_id ? self.first.default_skin : Skin.default) : Skin.default
+    default_skin=Rails.cache.fetch(default_skin_cache_key){Skin.default}
+    self.first ? (self.first.default_skin_id ? self.first.default_skin : default_skin) : default_skin
```

Comment:
Ternary operators must not be nested. Prefer `if`/`else` constructs instead.<br>Redundant `self` detected.